Updates for hawthorn upgrade

lgp171188 opened the pull request:
Add changes from https://github.com/open-craft/openedx-deployment/commit/cca980973364bf1136640e5649dd93f31de26296#diff-40c660c4b60b66b30b41740d0dd05ea3

and applies upstream change: https://github.com/edx/configuration/pull/4912

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

Commit:
Updates for hawthorn upgrade

lgp171188 commented on playbooks/analytics_sandbox.yml:
```diff
@@ -19,13 +19,11 @@
     - role: nginx
       nginx_sites:
         - insights
-        - analytics_api
```
@pomegranited, this line was not present in https://github.com/open-craft/openedx-deployment/commit/cca980973364bf1136640e5649dd93f31de26296#diff-40c660c4b60b66b30b41740d0dd05ea3. So I have removed it.

pomegranited replied:
Yep that's ok -- epodx don't expose their analytics API to the web, they use SSH tunnels.

pomegranited replied:
Sorry, I spoke too soon..  we do need that `analytics_api` line to exist under `nginx_sites`, and as a role on its own.  

epodx *do* expose their analytics API via nginx, though they use SSH tunnels to restrict access to the port itself.  I tested this by following the instructions in the secure repo docs for [accessing the Analytics API](https://github.com/hks-epod/epodx/blob/master/docs/api.md#hks-epodx---epodx-analytics-api).

lgp171188 replied:
@pomegranited, I will fix this and redeploy insights after the LMS redeployment is done and everything is working. :+1:

pomegranited reviewed (changes requested):
See [comment about analytics_api](https://github.com/hks-epod/configuration/pull/6/files#r243488242).

Commit:
Fix issue with rubygems 3.0.0 by pinning it to 2.7.8

Commit:
Remove demo role from the edx_sandbox.yml playbook

Commit:
Add the 'analytics_api' nginx site

Commit:
Move the analytics_api nginx site setup below analytics_api role

This is because the analytics_api role uses the 'edx_django_service'
role to setup the app and it uses a default app.j2 template instead of
the analytics_api.j2 nginx configuration template defined in the nginx
role which we want.

pomegranited reviewed (approved):
:+1: Thanks @lgp171188 !

- [x] I tested this by verifying the newly deployed analytics and edxapp instances are running the required services, and deployed successfully.
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A, changes come from other PRs.
